strings/boyer_moore_algorithm.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * \file
+ * \brief The [Boyer-Moore-Algorithm]
+ * (https://en.wikipedia.org/wiki/Boyer%E2%80%93Moore_string-search_algorithm)
+ * for finding a pattern within a piece of text with complexity O(n / m)
+ *
+ * 1. Boyer Moore is a combination of the following two approaches.
+ *     -> Bad Character Heuristic
+ *     -> Good Suffix Heuristic
+ * Both of the above heuristics can also be used independently to search a
+ * pattern in a text.
+ * 2. The idea of bad character heuristic is simple. The character of the text 
+ * which doesn’t match with the current character of the pattern is called the Bad 
+ * Character. Upon mismatch, we shift the pattern until – 
+ *     -> he mismatch becomes a match 
+ *     -> Pattern P moves past the mismatched character.
+ */
+
+#include <iostream>
+#ifdef _MSC_VER
+#include <string>  // use this for MS Visual C++
+#else
+#include <cstring>
+#endif
+using namespace std;
```

Comment:
Do not use `using namespace std;`. Use `using std::cin`, `using std::cout`, etc. for each function.